src/Invoice.php

```diff
@@ -26,7 +26,7 @@
  * @todo Add documentation and typehints to all properties.
  * @todo Add support for VatLines.
  */
-class Invoice
+class Invoice implements MatchReferenceInterface
```

Comment:
Perhaps it would be better to have a method `getMatchReference(): MatchReferenceInterface` on the `Invoice` class instead of implementing the interface directly. This way the `Invoice` class doesn't get certain public methods that don't make a lot of sense out of the `MatchReferenceInterface` context, for example, `$invoice->getLineId()` does not make a whole lot of sense on its own.

You can just return a `new MatchReference` from that `getMatchReference()` method as that implements the interface.

Reply:
Updated.